sdks/java/core/src/test/java/org/apache/beam/sdk/io/PubsubUnboundedSourceTest.java

```diff
@@ -272,7 +272,7 @@ public void readManyMessages() throws IOException {
       String recid = String.format("recordid_%d", messageNum);
       String ackId = String.format("ackid_%d", messageNum);
       incoming.add(new IncomingMessage(data.getBytes(), messageNumToTimestamp(messageNum), 0,
-                                       ackId, recid.getBytes()));
+                                       ackId, recid));
     }
     setupOneMessage(incoming);
 
```

Comment:
Note there's nothing here which tests the record dedup logic since it's upstream of the UnboundedSource transform.
But can't hurt to confirm the record id is random.